### PR TITLE
build: remove python fallback in overlay builds

### PR DESCRIPTION
**Pull request by compnerd**
Remove the Python 2 fallback from the standalone overlay build.  This
is part of the migration to Python 3.

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves SR-NNNN.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Review by tbkka — APPROVED** [review]
I think we're ready for this.

**Comment by shahmishal** [comment]
Please hold off from merging this PR, I would like to run additional testing on this. Thanks!

**Comment by compnerd** [comment]
This is associated with #32496 but both can be merged individually.

**Review by shahmishal — APPROVED** [review]
LGTM! Feel free to merge this when you'r ready!

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Pull request merged**
